### PR TITLE
Add license file for project

### PR DESCRIPTION
**Pull request by tuteng**
* Add license File for Dependency(Include backend and frontend)

### CODE REVIEW

**Comment by tuteng** [comment]
@eolivelli PTAL

File: distribution/NOTICE.bin.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+sqlite-jdbc-3.21.0.1.jar
+
+This product includes the following softwares developed by David Crawshaw.
+See LICENSE.zentus file.
+
+And also, NestedVM (Apache License Version 2.0) is used inside sqlite-
+
+----------------------------------------------------------------------------------------------------
+
+Apache Pulsar
```
Comment (sijie) [inline distribution/NOTICE.bin.txt]: Move pulsar section to the top and move `sqlite-jdbc` to the bottom.

**Comment by eolivelli** [comment]
> In the LICENSE.bin.txt, you might need to pointer to the license file for Non ALv2 dependencies.

This is indeed needed even for all of the projects that are ALv2 licensed but they do no come from ASF (like HerdDB).
see http://www.apache.org/dev/licensing-howto.html#alv2-dep

File: distribution/NOTICE.bin.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Copyright 2017-2019 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+----------------------------------------------------------------------------------------------------
+
+sqlite-jdbc-3.21.0.1.jar
```
Comment (eolivelli) [inline distribution/NOTICE.bin.txt]: this is no more needed, we could exclude it from the binary release

**Comment by tuteng** [comment]
> After writing my comments I have re-read the diff.
> I think this change is good.
> 
> It is an huge work !
> 
> I wonder if in the future it is possible to generate part of the contents automatically or at least change them.
> Gradle/Maven dependencies are easy to check but NPM/JS stuff is really hard as in my experience third party dependencies are changing very quickly.

@eolivelli Thank you very much to review. You are right, this work should be automated, and I will try to automate it if possible in the future. Currently, for node's packages, I use `npm show package-name` to obtain its license information. most of the packages can normally obtain license information, but there is still a small part that needs to be processed manually. I need some time to consider how to improve this problem.

**Pull request merged**